core/github_handler.py

```diff
@@ -0,0 +1,101 @@
+import logging
+
+from config import settings
+from core.base_handler import BaseHandler
+from generators.github import GithubIssueGenerator
+from targets.github import Github
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class GithubHandler(BaseHandler):
+    def issue_handler(self):
+        logger.info("Started syncing scorecard statuses as issues for Github")
+        if not self.entities:
+            logger.info("No entities found, looking for left over open issues")
+            issue_search_result = Github().search_issue_by_labels(
+                ["Port", self.scorecard.get("title", "")],
+                settings.github_repository,
+                state="open",
+            )
+            if len(issue_search_result) > 0:
+                logger.info("No open issues found")
+            else:
+                logger.info(f"Found {len(issue_search_result)} open issues")
+                for issue in issue_search_result:
+                    Github().close_issue(
+                        issue["number"], issue, settings.github_repository
+                    )
+            return
+
+        logger.info("Searching for Github issues to create / update")
+
+        for entity in self.entities:
+            entity_scorecard = entity.get("scorecards", {}).get(
+                self.scorecard.get("identifier"), {}
+            )
+            rules_by_level = {}
+
+            # Grouping rules by levels
+            for rule in entity_scorecard.get("rules", []):
+                level = rule.get("level")
+                if level not in rules_by_level.keys():
+                    rules_by_level[level] = []
+                rules_by_level[level].append(rule)
+
+            for level in rules_by_level:
+                scorecard_level_completed = all(
+                    rule.get("status", "") == "SUCCESS"
+                    for rule in rules_by_level[level]
+                )
```

Comment:
this could've been more optimized by just creating scorecard_level_completed = True and when you go over each rule, if its not successful set it to False

Reply:
This doesn't happen in the rule iteration but in the level, and the run iterations happen per level, So I don't understand where do you see the optimization, could you add a suggestion?